R/plotconfiguration-boxwhisker.R

```diff
@@ -0,0 +1,88 @@
+#' @title BoxWhiskerPlotConfiguration
+#' @docType class
+#' @description  Plot Configuration for Box Whisker Plots
+#' @export
+BoxWhiskerPlotConfiguration <- R6::R6Class(
+  "BoxWhiskerPlotConfiguration",
+  inherit = PlotConfiguration,
+
+  public = list(
+    initialize = function(title = "Time Profile Plot",
+                              subtitle = paste("Date:", format(Sys.Date(), "%y-%m-%d")),
+                              ...) {
+      super$initialize(
+        title = title,
+        subtitle = subtitle,
+        ...
+      )
+    },
+
+    addBoxWhisker = function(plotObject, data, metaData, dataMapping) {
+      mapData <- dataMapping$checkMapData(data, metaData)
+
+      # Get the box plot quantiles from dataMapping
+      mapQuantiles <- dataMapping$getQuantiles(data)
+
+      # Convert the mapping into characters usable by aes_string
+      mapLabels <- getAesStringMapping(dataMapping)
+
+      plotObject <- plotObject + ggplot2::geom_boxplot(
+        data = mapQuantiles,
+        mapping = aes_string(
+          x = mapLabels$x,
+          ymin = mapLabels$yMin,
+          lower = mapLabels$lower,
+          middle = mapLabels$middle,
+          upper = mapLabels$upper,
+          ymax = mapLabels$yMax,

```

Comment:
is there a reason why we are using ` yMax` and not `ymax` (same for ymin).

Reply:
yeah let's stick to ggplot nomenclature since we are using everything else from them